test/test-suite/src/main.rs

```diff
@@ -528,6 +530,272 @@ fn test_idol_ssmarshal() {
     assert!(r.f == 1.0);
 }
 
+#[cfg(feature = "i2c-devices")]
+include!(concat!(env!("OUT_DIR"), "/i2c_config.rs"));
+
+#[cfg(feature = "i2c-devices")]
+task_slot!(I2C, i2c_driver);
+
+// Put the FRU ID tests into their own module, so it can be enabled with
+// a single cfg block
+#[cfg(feature = "fru-id-eeprom")]
+mod at24csw080 {
+    use super::*;
+    use drv_i2c_devices::at24csw080::Error;
+    use drv_i2c_devices::at24csw080::*;
+
+    const EEPROM_SIZE: u16 = 1024;
+    const PAGE_SIZE: u16 = 16;
+
+    pub(super) fn test_at24csw080() {
+        let i2c_task = I2C.get_task_id();
+        let dev =
+            At24csw080::new(i2c_config::devices::at24csw080_local(i2c_task)[0]);
+
+        // The FRU ID EEPROM is 1KByte.  Because we want to exercise the
+        // ability to rewrite the EEPROM (and have that persist through power
```

Comment:
Are you testing this by manually power cycling?

Reply:
Yup!